batch/batch/driver/pool.py

```diff
@@ -0,0 +1,469 @@
+            self.add_instance(instance)
+
+        self.task_manager.ensure_future(self.control_loop())
+
+        async def scheduler_relay():
+            self.scheduler_state_changed.set()
+            return True
+
+        self.task_manager.ensure_future(retry_long_running(
+            'scheduler_state_changed_relay_loop',
+            run_if_changed, self.global_scheduler_state_changed,
+            scheduler_relay))
+
+        await self.scheduler.async_init()
+
+    def shutdown(self):
+        try:
+            super().shutdown()
+        finally:
+            self.scheduler.shutdown()
+
+    def config(self):
+        return {
+            'name': self.name,
+            'worker_type': self.worker_type,
+            'worker_cores': self.worker_cores,
+            'worker_disk_size_gb': self.worker_disk_size_gb,
+            'worker_local_ssd_data_disk': self.worker_local_ssd_data_disk,
+            'worker_pd_ssd_data_disk_size_gb': self.worker_pd_ssd_data_disk_size_gb,
+            'standing_worker': self.standing_worker,
+            'standing_worker_cores': self.standing_worker_cores,
+            'max_instances': self.max_instances,
+            'max_live_instances': self.max_live_instances
+        }
+
+    async def configure(
+            self, worker_cores, worker_disk_size_gb,
+            worker_local_ssd_data_disk, worker_pd_ssd_data_disk_size_gb,
+            standing_worker, standing_worker_cores,
+            max_instances, max_live_instances):
+        await self.db.just_execute(
+            '''
+UPDATE pools
+SET worker_cores = %s, worker_disk_size_gb = %s,
+  worker_local_ssd_data_disk = %s, worker_pd_ssd_data_disk_size_gb = %s,
+  standing_worker = %s, standing_worker_cores = %s,
+  max_instances = %s, max_live_instances = %s
+WHERE name = %s;
+''',
+            (worker_cores, worker_disk_size_gb,
+             worker_local_ssd_data_disk, worker_pd_ssd_data_disk_size_gb,
+             standing_worker, standing_worker_cores,
+             max_instances, max_live_instances, self.name))
+
+        self.worker_cores = worker_cores
+        self.worker_disk_size_gb = worker_disk_size_gb
+        self.worker_local_ssd_data_disk = worker_local_ssd_data_disk
+        self.worker_pd_ssd_data_disk_size_gb = worker_pd_ssd_data_disk_size_gb
+        self.standing_worker = standing_worker
+        self.standing_worker_cores = standing_worker_cores
+        self.max_instances = max_instances
+        self.max_live_instances = max_live_instances
+
+    def adjust_for_remove_instance(self, instance):
+        super().adjust_for_remove_instance(instance)
+        if instance in self.healthy_instances_by_free_cores:
+            self.healthy_instances_by_free_cores.remove(instance)
+
+    def adjust_for_add_instance(self, instance):
+        super().adjust_for_add_instance(instance)
+        if (instance.state == 'active'
+                and instance.failed_request_count <= 1):
```

Comment:
weird indentation here. Maybe but `and` on the previous line?

Reply:
I think that indentation is correct, but I moved the and to the line above.

Reply:
pylint complains if I change this.